monitoring/health_monitor.py

```
@@ -0,0 +1,341 @@
+
+            if response.status == service_config['expected_status']:
+                result['status'] = 'healthy'
+
+                # For health endpoints, try to parse the response
+                if 'health' in service_config['url']:
+                    try:
+                        data = json.loads(response.read().decode('utf-8'))
+                        if data.get('status') == 'degraded':
+                            result['status'] = 'degraded'
+                            result['error'] = data.get('message', 'Service degraded')
+                    except:
```

Ruff (0.15.5)

[error] Do not use bare `except`

(E722)